### PR TITLE
Define build failure conditions on scrutinizer

### PR DESCRIPTION
**Pull request by lcobucci**
So that @Ocramius can spend more time drinking and playing 😜

### CODE REVIEW

File: .scrutinizer.yml [hunk 1]
```diff
@@ -9,27 +9,117 @@ before_commands:
...
-    sensiolabs_security_checker: true
+
+filter:
+    excluded_paths:
+        - docs
+        - tools
+
+checks:
+    php:
+        code_rating: true
```
Comment (Ocramius) [inline .scrutinizer.yml]: Urgh, do we really want to maintain this infinite list of flags? :-(

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .scrutinizer.yml (hunk 1 above)
Comment (lcobucci) [inline .scrutinizer.yml]: Some are set by default, some are not... and those keep changing 😢

Comment (Ocramius) [inline .scrutinizer.yml]: Then let's please not go in this direction :-\

Comment (Ocramius) [inline .scrutinizer.yml]: The fact that it keeps changing is precisely the problem: if there was an XSD I would be OK with this, but not with some YAML cruft.

Comment (lcobucci) [inline .scrutinizer.yml]: So you want to remove the default ones from the config, right?

Comment (lcobucci) [inline .scrutinizer.yml]: I mean, since we don't have control over the format...

Comment (Ocramius) [inline .scrutinizer.yml]: The point is having *LESS* flags to maintain. Let's keep all the defaults we can keep, and only override the ones we need to override.

This project will never reach "object calisthenics" quality within our lifetime, so it's useless to use a code quality analysis tool to drive improvements: we are good at reviewing stuff.

Comment (lcobucci) [inline .scrutinizer.yml]: The point is to also keep https://scrutinizer-ci.com/g/doctrine/doctrine2/issues/master updated with things that we consider to be improved (and which people could just pick up).

I mean, the defaults are fine but they're too generic... maybe we could find a middle ground?

**Comment by Ocramius** [comment]
I'd just keep the defaults for now: the tool is interesting for finding low
hanging fruit issues.

**Comment by lcobucci** [comment]
@Ocramius  ok, but do you agree with adding the failing conditions?

**Comment by Ocramius** [comment]
I'm fine with additional failure conditions, yes. Just overwhelmed with the
size of the diff

**Comment by lcobucci** [comment]
@Ocramius have fun 😉

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Pull request merged**